### ISSUE TITLE
markdown content not able to be rendered: Error: Error building site

### ISSUE THREAD

**Issue by luxcgo** [gohugoio/hugo#8727]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.83.1+extended darwin/arm64 BuildDate=unknown
</pre>

### Does this issue reproduce with the latest release?
yes
below is the file that triggered the bug, I'm using typora editor
[temp.txt](https://github.com/gohugoio/hugo/files/6759387/temp.txt)
```
Start building sites …
goroutine 64 [running]:
runtime/debug.Stack(0x105f7c638, 0x106b2df00, 0x1400014a540)
	runtime/debug/stack.go:24 +0x88
github.com/gohugoio/hugo/markup/goldmark.(*goldmarkConverter).Convert.func1(0x14000a10000, 0x140004f6840, 0x20a, 0x28a, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/gohugoio/hugo/markup/goldmark/convert.go:239 +0x188
panic(0x105e1b020, 0x14000a00ab0)
	runtime/panic.go:965 +0x14c
github.com/yuin/goldmark/text.(*Segment).Value(0x14000551248, 0x140004f6840, 0x20a, 0x28a, 0x19, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.5/text/segment.go:48 +0x30c
github.com/yuin/goldmark-highlighting.(*HTMLRenderer).renderFencedCodeBlock(0x1400086c0e0, 0x105f74d38, 0x14000b8ea00, 0x140004f6840, 0x20a, 0x28a, 0x105f87e60, 0x140006a6000, 0x201, 0x3, ...)
	github.com/yuin/goldmark-highlighting@v0.0.0-20200307114337-60d527fdb691/highlighting.go:465 +0x698
github.com/yuin/goldmark/renderer.(*renderer).Render.func2(0x105f87e60, 0x140006a6000, 0x1, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.5/renderer/renderer.go:166 +0xdc
github.com/yuin/goldmark/ast.walkHelper(0x105f87e60, 0x140006a6000, 0x14000551650, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.5/ast/ast.go:492 +0x40
github.com/yuin/goldmark/ast.walkHelper(0x105f88520, 0x140006a41b0, 0x14000551650, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.5/ast/ast.go:498 +0x148
github.com/yuin/goldmark/ast.walkHelper(0x105f88400, 0x140006a4090, 0x14000551650, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.5/ast/ast.go:498 +0x148
github.com/yuin/goldmark/ast.walkHelper(0x105f87c20, 0x140006a4000, 0x14000551650, 0x105f74d38, 0x14000b8ea00, 0x1)
	github.com/yuin/goldmark@v1.3.5/ast/ast.go:498 +0x148
github.com/yuin/goldmark/ast.Walk(...)
	github.com/yuin/goldmark@v1.3.5/ast/ast.go:487
github.com/yuin/goldmark/renderer.(*renderer).Render(0x140002971d0, 0x105f44798, 0x14000b8ea00, 0x140004f6840, 0x20a, 0x28a, 0x105f87c20, 0x140006a4000, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.5/renderer/renderer.go:161 +0xdc
github.com/gohugoio/hugo/markup/goldmark.(*goldmarkConverter).Convert(0x14000a10000, 0x140004f6840, 0x20a, 0x28a, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/gohugoio/hugo/markup/goldmark/convert.go:265 +0x488
github.com/gohugoio/hugo/hugolib.(*pageContentOutput).renderContentWithConverter(0x1400087c540, 0x105f565d0, 0x14000a10000, 0x140004f6840, 0x20a, 0x28a, 0x14000997a01, 0x104faa7a0, 0x1400087c614, 0x14000997a18, ...)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:378 +0xa8
github.com/gohugoio/hugo/hugolib.(*pageContentOutput).renderContent(0x1400087c540, 0x140004f6840, 0x20a, 0x28a, 0x140004ee601, 0x0, 0x105f566e8, 0x1400063e270, 0x45)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:374 +0x58
github.com/gohugoio/hugo/hugolib.newPageContentOutput.func1(0x0, 0x0)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:126 +0x848
github.com/gohugoio/hugo/hugolib.newPageContentOutput.func2(0x105f69040, 0x14000770de0, 0x1057d7e00, 0x1, 0x1, 0x14000642600)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:205 +0x28
github.com/gohugoio/hugo/lazy.(*Init).withTimeout.func1(0x1400033ba60, 0x105f69040, 0x14000770de0, 0x14000770e40)
	github.com/gohugoio/hugo/lazy/init.go:176 +0x38
created by github.com/gohugoio/hugo/lazy.(*Init).withTimeout
	github.com/gohugoio/hugo/lazy/init.go:175 +0xd0
Built in 38 ms
Error: Error building site: "/Users/lucas/Documents/blog/myhugoblog/content/posts/go test.md:1:1": [BUG] goldmark: runtime error: slice bounds out of range [343:341]: create an issue on GitHub attaching the file in: /var/folders/23/6dp1p65d3ng56j320xxr2wfh0000gn/T/hugo_bugs/goldmark_927bd71bfc7c802ffbb4d449f9e1a8b6.txt
```

**Comment by bep** [comment]
Thanks for the report. This looks like it could be an issue with Goldmark. Could you create an issue at https://github.com/yuin/goldmark/issues and link back to this issue so we could track it?

**Comment by luxcgo** [comment]
Thanks for your reply! I checked goldmark issues. It seems the problem is already solved in v1.3.9 release version. We only need to update it. yuin/goldmark/issues/229
